### DISCUSSION TITLE
Traffic between web and worker can be encrypted ?

### DISCUSSION THREAD

**Discussion by gowrisankar22** [concourse/concourse#5857] [Help & Support]
Hello,

As per https://concourse-ci.org/teams-caveats.html  

1. Traffic to and from the workers is currently unencrypted and unauthorized.
2. Traffic from worker to/from Webnode is again unencrypted.

**The concern:**  

1. Can I sniff communication of the other containers on this worker to/from Web nodes, from my container (it is plain HTTP at the end)? 
2. I do realize that this might not be the only issue (considering other plain internal HTTP communications) considering the privilege escalation option we cannot forbid. 

Is there a possible option to mitigate this?

BRs, Gowrisankar

**Comment by jamieklassen** [comment]
I'm not totally sure what you mean by "the other containers on this worker", but there is a good tcpdump example showing some plaintext communication between web and worker: https://github.com/concourse/concourse/issues/2415

**Reply by gowrisankar22** [reply]
@jamieklassen I would like to know mainly, can we encrypt the communication between the web and workers?

i tried with `tcpdump -i eth0 'port 7777' -X -v` i see that `0` packets received all the time. I will continuously try

```
 tcpdump: listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes
0 packets captured
0 packets received by filter
0 packets dropped by kernel
```

**Reply by jamieklassen** [reply]
I don't have any good ideas about how to do it, and I haven't heard of anyone successfully doing it. I just wanted to offer a link so you could learn more about your "can I sniff" question. I'm not going to help you debug your tcpdump thing.

One day we'll probably spend some time thinking about how to get this into the core product, either by doing our own TLS termination or suggesting some kind of proxy configuration similar to what @eedwards-sk suggested here: https://github.com/concourse/concourse/issues/2415#issuecomment-500513913
